Fix iceberg data migration test timeout

bashtanov opened the pull request:
The test is broken: it swallows errors, timeout ones in particular.
Make it propagate errors. To avoid timeouts:
- make sure verifier offline mode only waits for consumption not for anything in the querying thread
- reduce production rate
- allow more time to complete

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [x] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

bashtanov committed:
tests/datalake/verifier: do not swallow exceptions when waiting

bashtanov committed:
tests/datalake/verifier: offline mode improvements

- wait for consuming all messages regardless translation state
- avoid race conditions when stopping consumer

bashtanov committed:
tests/datalake/mount_unmount: remove sleep, reduce pace, give time

Production rate reduced as otherwise RPCN produces too much data while
unmounting, so it takes unreasonable time to complete.

Sleep removed as verifier is more robust.

Time limit increased as sometimes unmount takes more time and it takes
longer to get offline:
- offline mode waits for consuming till migrations blocking offset
- consume thread waits for query thread (limited comparison buffer)
- query thread may lag because translation lags

bashtanov committed:
tests/util/rpcn/counter_stream_config: make rate configurable

sometimes maximum throughput is not desired

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--50058c09-7ec3-4720-bf60-900acdcd6b22-->
<!--TEST RESULTS-->
<!--39716c5e-6012-423a-a891-c70a7974c2d3-->
<!--TEST RESULTS-->
<!--2f57f2e4-4101-4aef-bf13-e583277bb8d7-->
<!--TEST RESULTS-->
<!--5d96793c-20bb-4b76-84d7-0851aed1be35-->
<!--TEST RESULTS-->
<!--d8354b6f-9d80-49ff-a255-19413a37bc1f-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61767">build#61767</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed78-5b46-4f61-a4d1-c0b9f80c8fe8 | FLAKY | 1/3 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed78-5b46-4a7a-b743-4765ba61ffe9 | FLAKY | 1/2 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-07a0-4a9d-b85f-052e159a33fa | FLAKY | 1/2 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079e-47f9-8066-e78270290712 | FLAKY | 1/2 |
| rptest.tests.datalake.mount_unmount_test.MountUnmountIcebergTest.test_simple_unmount.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079f-4549-a7c5-cf74a8afb220 | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079f-43ff-85ad-4a407350be3a | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=10 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079f-43ff-85ad-4a407350be3a | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079e-47f9-8066-e78270290712 | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079f-43ff-85ad-4a407350be3a | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/61767#0194ed65-079f-43ff-85ad-4a407350be3a | FLAKY | 1/3 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61891">build#61891</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.availability_test.AvailabilityTests.test_recovery_after_catastrophic_failure | ducktape | https://buildkite.com/redpanda/redpanda/builds/61891#01950c9d-c81b-4cd0-b9c8-8b55cd0d47f5 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61891#01950c9d-c81d-4026-9b9d-23174d15a298 | FLAKY | 1/2 |
| rptest.tests.delete_records_test.DeleteRecordsTest.test_delete_records_concurrent_truncations.cloud_storage_enabled=True.truncate_point=start_offset | ducktape | https://buildkite.com/redpanda/redpanda/builds/61891#01950c9d-c81c-4acb-a996-8092df40b022 | FLAKY | 1/2 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move.replication_factor=3.unclean_abort=True.recovery=no_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/61891#01950c9d-c81b-4cd0-b9c8-8b55cd0d47f5 | FLAKY | 1/2 |
| storage_single_thread_rpunit.storage_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/61891#01950c5a-10e6-4d7d-8288-a8b7815d2517 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61893">build#61893</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61893#01950e4e-95ab-4976-b770-c4685c51f5e9 | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61893#01950e68-4471-4097-9721-039f49ff8226 | FLAKY | 1/5 |
| storage_single_thread_rpunit.storage_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/61893#01950e09-c5a2-43c5-9445-1a61bce8346c | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62179">build#62179</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.audit_log_test.AuditLogTestsAppLifecycle.test_app_lifecycle | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#019537ec-114d-456f-a43f-7b040a640bca | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#019537ec-114d-456f-a43f-7b040a640bca | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#01953806-f982-43a8-a429-b1e51cb1a5cc | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#01953806-f981-4d64-9151-4adfe2927e56 | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#01953806-f983-4b67-acd1-47fe356b98cd | FLAKY | 1/2 |
| rptest.tests.write_caching_fi_e2e_test.WriteCachingFailureInjectionE2ETest.test_crash_all_with_consumer_group | ducktape | https://buildkite.com/redpanda/redpanda/builds/62179#019537ec-114d-456f-a43f-7b040a640bca | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62203">build#62203</a></summary>


| test_id                                                                                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   |
|:-------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.partition_movement_test.SIPartitionMovementTest.test_shadow_indexing.num_to_upgrade=0.cloud_storage_type=CloudStorageType.ABS | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62203#01953969-87ba-4b89-bd7b-92c922ea417e | FLAKY         | 1/2      |
</details>
<!--/body-->

bashtanov commented:
Meh. It worked locally. Will debug.

bashtanov committed:
tests/datalake/verifier: make buffer size configurable

1000 messages per second is not a crazy production rate, but a buffer
of 5000 messages will only keep 5 seconds worth, which is less than
unmount or translation delays

bashtanov commented:
/dt

andrwng commented on tests/rptest/tests/datalake/datalake_verifier.py:
```diff
@@ -47,7 +47,8 @@ def __init__(self,
                  compacted: bool = False,
-                 table_override: Optional[str] = None):
+                 table_override: Optional[str] = None,
+                 buffer=5000):
```
nit: maybe call this `max_buffered_msgs` or somesuch?

andrwng commented on tests/rptest/tests/datalake/mount_unmount_test.py:
```diff
@@ -102,12 +102,15 @@ def test_simple_unmount(self, cloud_storage_type):
                              config=self.avro_stream_config(
-                                 self.TOPIC_NAME, "verifier_schema", 1000000))
+                                 self.TOPIC_NAME, "verifier_schema", 1000000,
+                                 1))
```
nit: maybe pull 1 out into some `low_interval_ms` and explain in a comment why it's necessary?

andrwng commented on tests/rptest/tests/datalake/mount_unmount_test.py:
```diff
@@ -102,12 +102,15 @@ def test_simple_unmount(self, cloud_storage_type):
+        verifier = DatalakeVerifier(self.redpanda,
+                                    self.TOPIC_NAME,
+                                    self.dl.spark(),
+                                    buffer=50000)
```
nit: maybe pull 50000 out into some `high_buffered_msgs` and explain in a comment why it's necessary?

andrwng commented on tests/rptest/tests/datalake/mount_unmount_test.py:
```diff
@@ -127,8 +130,7 @@ def test_simple_unmount(self, cloud_storage_type):
         connect.stop_stream("ducky_stream", should_finish=False)
-        time.sleep(1)  # just it case: let verifier consume remaining messages
-        verifier.go_offline()
+        verifier.go_offline(600)
```
nit: comment explaining why this is necessary?

andrwng commented on tests/rptest/tests/datalake/mount_unmount_test.py:
```diff
@@ -73,14 +73,14 @@ def __init__(self, test_context):
             include_query_engines=[QueryEngineType.SPARK])
 
-    def avro_stream_config(self, topic, subject, cnt=3000):
+    def avro_stream_config(self, topic, subject, cnt=3000, interval_ms=None):
```
>Production rate reduced as otherwise RPCN produces too much data while unmounting

Hmm I thought we blocked writes while we unmounted. Are we sure we eventually actually quiesce? The tests I saw were hanging around for 12 minutes without completing -- I don't imagine unmount would take _that_ long, would it?

bashtanov replied:
Yes we do, but it takes some time since we block it.
The 12 minutes you saw was a combination of many problems, swallowed errors in particular -- see 1st commit.

bashtanov committed:
tests/datalake/mount_unmount: disable debug mode

to avoid writing too much data

bashtanov commented:
failures unrelated
